Server/mods/deathmatch/logic/luadefs/CLuaTeamDefs.cpp

```
@@ -67,282 +66,83 @@ void CLuaTeamDefs::AddClass(lua_State* luaVM)
     lua_registerclass(luaVM, "Team", "Element");
 }
 
-int CLuaTeamDefs::SetPlayerTeam(lua_State* luaVM)
+CTeam* CLuaTeamDefs::CreateTeam(lua_State* lua, std::string name, std::uint8_t red, std::uint8_t green, std::uint8_t blue) noexcept
```

This method isn't backward compatible. Because originally `createTeam` returns `false` on failure. But after this PR it will return `nil`.

---

To solve this problem, what would be the best solution?

---

`std::variant<CTeam, bool> CreateTeam(...)`

---

```c++
if (!team)
        return nullptr;
```

You can throw an error here. IIRC `ArgumentParserWarn` returns false and shows your message.

---

> ```c++
> if (!team)
>         return nullptr;
> ```
> 
> You can throw an error here. IIRC `ArgumentParserWarn` returns false and shows your message.

This should work. But originally these functions return `false` silently. For example, `getTeamFromName `can be used for checking for the presence of a team and error messages would be too intrusive.